### PR TITLE
Add `HyperlinkRelated` field for SQLAlchemy integration.

### PR DESCRIPTION
**Pull request by jmcarp**
Second attempt at https://github.com/marshmallow-code/marshmallow-sqlalchemy/pull/10.

Note: depends on https://github.com/marshmallow-code/marshmallow-sqlalchemy/pull/8 and https://github.com/marshmallow-code/marshmallow/pull/230.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,9 +13,11 @@ env:
...
 
 install:
-  - pip install -U -r dev-requirements.txt
-  - pip install -U .
-  - pip install -U marshmallow"$MARSHMALLOW_VERSION" --pre
+  - travis_retry pip install -U -r dev-requirements.txt
+  - travis_retry pip install -U .
+  - travis_retry pip install -U marshmallow"$MARSHMALLOW_VERSION" --pre
+  - travis_retry pip install -U git+https://github.com/jmcarp/marshmallow@set-field-attrs
+  - travis_retry pip install -U git+https://github.com/jmcarp/marshmallow-sqlalchemy@refactor-related
```
Comment (sloria) [inline .travis.yml]: The supporting PRs have been merged so these can be changed to the `dev` branches.

File: flask_marshmallow/sqla.py [hunk 2]
```diff
@@ -41,31 +42,30 @@ class ModelSchema(msqla.ModelSchema, Schema):
...
+    def __init__(self, endpoint, url_key='id', external=False, **kwargs):
+        super(HyperlinkRelated, self).__init__(**kwargs)
+        self.endpoint = endpoint
+        self.url_key = url_key
+        self.external = external
 
-class HyperlinkModelSchema(msqla.ModelSchema):
-    """A `ModelSchema <marshmallow_sqlalchemy.ModelSchema>` that serializes relationships
-    to hyperlinks. Related models MUST have a ``url`` attribute or property (unless the
-    ``MARSHMALLOW_LINK_ATTRIBUTE`` app config option is set).
```
Comment (sloria) [inline flask_marshmallow/sqla.py:65]: If we're getting rid of this, then the `MARSHMALLOW_LINK_ATTRIBUTE` on ~L121 can be removed.

File: flask_marshmallow/sqla.py [hunk 3]
```diff
@@ -41,31 +42,30 @@ class ModelSchema(msqla.ModelSchema, Schema):
...
-    See `marshmallow_sqlalchemy.ModelSchema` for more details
-    on the `ModelSchema` API.
-    """
+    def _deserialize(self, value):
+        if self.external:
+            parsed = parse.urlparse(value)
+            value = parsed.path
+        endpoint, kwargs = self.adapter.match(value)
+        if endpoint != self.endpoint:
+            raise
```
Comment (sloria) [inline flask_marshmallow/sqla.py]: What is raised here?

Comment (jmcarp) [inline flask_marshmallow/sqla.py]: Um, I forgot to pick an exception. How about `ValidationError`?

Comment (sloria) [inline flask_marshmallow/sqla.py]: Yes, `ValidationError` seems correct.

**Comment by jmcarp** [comment]
Now with non-`RuntimeError` exceptions!

**Pull request merged**